logagg_collector/collector.py

```diff
@@ -0,0 +1,926 @@
+        >>> temp_dir_path = temp_dir.name
+        >>> trackfile = open(temp_dir_path+'/trackfiles.txt', 'w+'); trackfile.close()
+        >>> lc = collector = LogCollector('localhost', '1088', None, temp_dir_path, temp_dir_path)
+        >>> from pprint import pprint
+
+        >>> formatter = 'logagg.formatters.mongodb'
+        >>> fpath = '/var/log/mongodb/mongodb.log'
+        >>> line = 'some log line here'
+
+        >>> default_log = lc._assign_default_log_values(fpath, line, formatter)
+        >>> pprint(default_log) #doctest: +ELLIPSIS
+        {'data': {},
+         'error': False,
+         'error_tb': '',
+         'event': 'event',
+         'file': '/var/log/mongodb/mongodb.log',
+         'formatter': 'logagg.formatters.mongodb',
+         'host': '...',
+         'id': None,
+         'level': 'debug',
+         'raw': 'some log line here',
+         'timestamp': '...',
+         'type': 'log'}
+
+        >>> temp_dir.cleanup()
+        '''
+        return dict(
+            id=None,
+            file=fpath,
+            host=self.host,
+            formatter=formatter,
+            event='event',
+            data={},
+            raw=line,
+            timestamp=datetime.datetime.utcnow().isoformat(),
+            type='log',
+            level='debug',
+            error= False,
+            error_tb='',
+          )
+
+
+    def _delete_file(self, fpath):
+        '''
+        Move log file from logaggfs 'logs' directory
+        >>> import tempfile
+        >>> temp_dir = tempfile.TemporaryDirectory()
+        >>> temp_dir_path = temp_dir.name
+        >>> trackfile = open(temp_dir_path+'/trackfiles.txt', 'w+'); trackfile.close()
+        >>> lc = collector = LogCollector('localhost', '1088', None, temp_dir_path, temp_dir_path)
+
+        >>> log_file_dir = tempfile.TemporaryDirectory()
+        >>> log_file_path = log_file_dir.name + '/log_file.log'
+        >>> offset_file_path = log_file_path + '.offset'
+        >>> log_file = open(log_file_path, 'w+'); trackfile.close()
+        >>> offset_file = open(offset_file_path, 'w+'); offset_file.close() 
+        >>> import os
+        >>> os.path.isfile(log_file_path)
+        True
+        >>> os.path.isfile(offset_file_path)
+        True
+        >>> lc._delete_file(log_file_path)
+        >>> os.path.isfile(log_file_path)
+        False
+        >>> os.path.isfile(offset_file_path)
+        False
+
+        >>> temp_dir.cleanup()
+        >>> log_file_dir.cleanup()
+        '''
+        os.remove(fpath)
+        os.remove(fpath+'.offset')
+
+
+    @keeprunning(LOG_FILE_POLL_INTERVAL, on_error=utils.log_exception)
+    def _collect_log_files(self, log_files):
+        '''
+        Collect from log files in logaggfs 'logs' one by one
+        '''
+
+        L = log_files
+        # Sorted list of all the files for one pattern
+        fpaths = glob.glob(join(self.logaggfs.logs_dir, L['fpattern']))
+        fpaths = sorted(fpaths)
+
+        for f in fpaths:
```

Comment:
may a different name other than `f` would be good because usually, we use `f` for file pointer, but it is file name here.